queries.sql

```diff
@@ -9,3 +9,30 @@ select * from animals where neutered = 't';
 select * from animals where name != 'Gabumon';
 select * from animals where weight_kg <= 10.4 or weight_kg >= 17.3;
 
+/*working with TRANSACTIONS. */
+BEGIN;
+update animals set species = 'digimon' where name like '%mon';
+update animals set species = 'pokemon' where species ='';
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals where date_of_birth > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals SET weight_kg = weight_kg*-1;
+ROLLBACK TO SP1;
+UPDATE animals SET weight_kg = weight_kg*-1 WHERE weight_kg <0;
+COMMIT;
+
+/*Working with aggregate functions*/
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, MAX(escape_attempts) FROM animals GROUP BY neutered;
+SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-01-01' GROUP BY species;
```

Comment:
- [ ] **Required:** The date you have specified would begin 1990 jan 1st and end 2000 jan 1st but the remaining months of the year 2000 would be left out. Kindly update it to `2000-12-31` if it matches your format.